extensions/amp-video/0.1/amp-video.js

```diff
@@ -241,6 +243,11 @@ class AmpVideo extends AMP.BaseElement {
     listen(video, 'ended', () => {
       this.element.dispatchCustomEvent(VideoEvents.PAUSE);
     });
+    ['durationchange', 'timeupdate', 'seeking'].map(e => {
+      listen(video, e, () => {
```

Comment:
This creates 4 function instances when only 1 is necessary:

```js
array.map(e => { // This is function 1
  listen(video, e, () => {
    // This is created 3 times
  });
});
```

I see why you've done this since it's a lot of reusable code. But, we could implement this with a hoisted closure, and reap memory benefits:

```js
const dispatch = () => {
  // dispatch
};
listen(video, 'durationchange', dispatch);
listen(video, 'timeupdate', dispatch);
listen(video, 'seeking', dispatch);